File: packages/square-mobile-payments/ios/Plugin/Classes/Options/InitializeOptions.swift
import Foundation
import Capacitor

@objc public class InitializeOptions: NSObject {
    let locationId: String

    init(_ call: CAPPluginCall) throws {
        self.locationId = try InitializeOptions.getLocationIdFromCall(call)
    }

    private static func getLocationIdFromCall(_ call: CAPPluginCall) throws -> String {
        guard let locationId = call.getString("locationId") else {
            throw CustomError.locationIdMissing
        }
        return locationId
    }
}

File: packages/square-mobile-payments/android/src/main/java/io/capawesome/capacitorjs/plugins/squaremobilepayments/classes/options/InitializeOptions.java
package io.capawesome.capacitorjs.plugins.squaremobilepayments.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.squaremobilepayments.classes.CustomExceptions;

public class InitializeOptions {

    @NonNull
    private final String locationId;

    public InitializeOptions(@NonNull PluginCall call) throws Exception {
        this.locationId = InitializeOptions.getLocationIdFromCall(call);
    }

    @NonNull
    public String getLocationId() {
        return locationId;
    }

    @NonNull
    private static String getLocationIdFromCall(@NonNull PluginCall call) throws Exception {
        String locationId = call.getString("locationId");
        if (locationId == null) {
            throw CustomExceptions.LOCATION_ID_MISSING;
        }
        return locationId;
    }
}
